third_party/terraform/resources/resource_compute_router_interface.go

```diff
@@ -20,6 +20,8 @@ func resourceComputeRouterInterface() *schema.Resource {
 			State: resourceComputeRouterInterfaceImportState,
 		},
 
+		CustomizeDiff: diffOneOfCheck,
```

Comment:
I would prefer that this either allow passing in the list of fields to be validated or renamed to something that indicates this is only a validation function for `compute_router_interface` only.